distributed/client.py

```diff
@@ -1371,17 +1371,22 @@ def _dec_ref(self, key):
             self.refcount[key] -= 1
             if self.refcount[key] == 0:
                 del self.refcount[key]
-                self._release_key(key)
+                self._release_key(key, f"client-release-key-{time()}")
 
-    def _release_key(self, key):
+    def _release_key(self, key, stimulus_id: str):
```

Comment:
To expand on my earlier comments, to me this would imply a ContextVar for the Client, which should be distinct from the Scheduler. My view on the situation is that ultimately Clients, Schedulers and Workers operate in their own context -- this is certainly the case in a cluster situation.

As such the ContextVar stimulus_id approach *only* applies to the Scheduler in this PR

Reply:
I'd like to leave this for an extension PR